### PR TITLE
Cancellations and Recommendations

### PR DESCRIPTION
**Pull request by OutbreakSource**
-Recommendations method is created, uses SQL queries to find relative orders from other customers to the current order being processed in emails. Returns a string which then is used to create a second email to be sent for customers to be sent. Only has to be refactored if the string is blank due to non-common orders to make it not send out an email at all.

-Cancellation now works. When cancelling the order, it changes the status of the order on the database to '-1' to show that it was cancelled. You just have to send an email back to mycrowsawftburner with the subject 'cancel' and the order id you were given when you ordered initially.

### CODE REVIEW

**Comment by GalacticWafer** [comment]
Daniel,
Overall, 98% of the things we need to do in this class seem to be complete! Just a few code cleanup requests to finalize work in this class:
	- It looks like we don't need to import `java.time.DayOfWeek` and `java.time.LocalDate` anymore in the Emailer class, so we can delete those.
	- Let's get rid of the `private Session session` field. We can just call `credentials.getSession()` instead of "pulling it out" of the credentials object. 
	- We should also have some basic Javadoc comment for the new `recommendProducts` method as well as the `createNewMessage` method and `class Emailer` itself.		

Great job!

**Review by GalacticWafer — APPROVED** [review]

**Review by Hamadesigns — APPROVED** [review]
changes reviewed and approved,

File: src/main/java/customerrelationsmanagement/Emailer.java [hunk 1]
```diff
@@ -159,44 +179,79 @@ public void processEmails(Crud crud)
...
 							String orderId = s[0];
 							Object[][] records = crud.getRecords(
-							 "SELECT * FROM sales where order_id = '"
-							 + orderId + "'");
+							 "SELECT * FROM statused_sales where order_id = '"
+							 + orderId + "'" + " and order_status = " +
+							 Order.PROCESSED);
 							System.out.println(
-							 "The following product purchases should be " +
+							 " The following product purchases should be " +
```
Comment (dbowling21) [inline src/main/java/customerrelationsmanagement/Emailer.java:186]: now that testing is complete this line printing to the console isn't really necessary

File: src/main/java/customerrelationsmanagement/Emailer.java [hunk 2]
```diff
@@ -159,44 +179,79 @@ public void processEmails(Crud crud)
...
+							if(records.length != 0) {
+								crud.update(
+								 "update statused_sales set order_status = " +
+								 "-1" +
+								 " " +
+								 "where order_id = '" +
+								 orderId + "'");
+								sendMail(
+								 email, "Cancellation",
+								 " The following product purchases should be" +
```
Comment (dbowling21) [inline src/main/java/customerrelationsmanagement/Emailer.java:205]: Change the email message from should be canceled to have been canceled. Or something that gives customers more confidence

**Review by dbowling21 — APPROVED** [review]
Like Malcolm said a few areas to clean up but this class is looking pretty close to finished. The last remaining is to handle incorrect inputs it looks like.  Otherwise, the recommendation system looks like a pretty strong implementation, nice work.

File: src/main/java/customerrelationsmanagement/Emailer.java (hunk 2 above)
Comment (OutbreakSource) [reply src/main/java/customerrelationsmanagement/Emailer.java:205]: good point

**Pull request merged**